src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3560,10 +3560,31 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 												bool staticOnly = methodParams [0].Kind == ValueNodeKind.Null;
 												// TODO: Change this as needed after deciding if we are to keep all fields/properties on a type
 												// that is accessed via reflection. For now, let's only keep the field/property that is retrieved.
-												if (calledMethod.Name [0] == 'P')
-													MarkPropertiesFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, stringValue.Contents, staticOnly);
-												else
-													MarkFieldsFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, stringValue.Contents, staticOnly);
+												if (calledMethod.Name [0] == 'P') {
+													MarkPropertiesOnTypeHierarchy (ref reflectionContext, systemTypeValue.TypeRepresented,
+														(property) => {
+															if (property.Name != stringValue.Contents) return false;
+
+															if (staticOnly) {
+																if (property.GetMethod != null && !property.GetMethod.IsStatic) return false;
+																if (property.SetMethod != null && !property.SetMethod.IsStatic) return false;
+															}
+
+															return true;
+														});
+												} else {
+													MarkFieldsOnTypeHierarchy (ref reflectionContext, systemTypeValue.TypeRepresented,
```

Comment:
Just a thought: With this the func you pass in requires a closure (the `staticOnly` variable) which means each call here will allocate a new object. If we move the check for static only inside the method and pass it as a parameter we would avoid the allocation. I know it's not as nice necessarily, but it will be faster.